src/main/java/org/testng/TestRunner.java

```diff
@@ -892,10 +892,14 @@ private void logStart() {
    *     finish
    */
   private void fireEvent(boolean isStart) {
-    for (ITestListener itl : m_testListeners) {
-      if (isStart) {
+    if(isStart) {
+      for (ITestListener itl : m_testListeners) {
         itl.onStart(this);
-      } else {
+      }
+    } else {
+      List<ITestListener> m_testListeners_Reverted = new ArrayList<>(m_testListeners);
```

Comment:
Prefixing the `m_` has been something that has been historically followed in the TestNG codebase. But apart from that, IMO we dont use `_` to separate words in variable names. Can you please help rename this to `m_testListenersReversed` instead of `m_testListeners_Reverted` ?

Reply:
Done.

Reply:
Fyi, `m_xxx` is the very old coding convention from Sun